src/main/java/net/minecraftforge/client/gui/widget/ForgeSlider.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Minecraft Forge - Forge Development LLC
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.gui.widget;
+
+import net.minecraft.client.gui.components.AbstractSliderButton;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.util.Mth;
+import org.lwjgl.glfw.GLFW;
+
+import java.text.DecimalFormat;
+
+/**
+ * Slider widget implementation which allows inputting values in a certain range with optional step size.
+ */
+public class ForgeSlider extends AbstractSliderButton
+{
+    protected Component prefix;
+    protected Component suffix;
+
+    protected double minValue;
+    protected double maxValue;
+
+    /** Allows input of discontinuous values with a certain step */
+    protected double stepSize;
+
+    protected boolean drawString;
+
+    private final DecimalFormat format;
+
+    /**
+     * @param precision Only used when {@code stepSize} is 0
+     */
+    public ForgeSlider(int x, int y, int width, int height, Component prefix, Component suffix, double minValue, double maxValue, double currentValue, double stepSize, int precision, boolean drawString)
+    {
+        super(x, y, width, height, TextComponent.EMPTY, 0D);
+        this.prefix = prefix;
+        this.suffix = suffix;
+        this.minValue = minValue;
+        this.maxValue = maxValue;
+        this.stepSize = Math.abs(stepSize);
+        this.value = this.snapToNearest((currentValue - minValue) / (maxValue - minValue));
+        this.drawString = drawString;
+
+        if (stepSize == 0D)
+        {
+            precision = Math.min(precision, 4);
```

Comment:
Can we note this in the javadoc so that people aren't confused when >4 fails to work

Reply:
Yeah, sure, was a late addition, so I forgot, will add the docs later.